ducktape/tests/test.py

```diff
@@ -143,6 +144,7 @@ def __init__(self, session_context, module=None, cls=None, function=None, inject
         self.cls = cls
         self.function = function
         self.injected_args = injected_args
+        self.ignore = ignore  # Don't run this test if ignore is True
```

Comment:
I'd keep this class as minimal as possible and just allow annotations to add attributes as they need to. This does require a bit more care when using the attributes, but decouples the details of the implementation of the annotations from the core TestContext object.

The same could probably also hold for `injected_args`. I don't think it *needs* to always be set on this object. The major difference between the two, however, which would make removing `injected_args` a more significant change, is that `injected_args` is used to generate a bunch of derived info like the output directory and full test name.